src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1244,11 +1246,13 @@ private void ParseModifiers(SyntaxListBuilder tokens, bool forAccessors)
                         }
 
                     case DeclarationModifiers.Async:
-                        if (!ShouldAsyncBeTreatedAsModifier(parsingStatementNotDeclaration: false))
+                        if (!ShouldAsyncBeTreatedAsModifier(parsingStatementNotDeclaration: false, alreadyHadAsyncModifier: alreadyHadAsyncModifier))
```

Comment:
i am a little confsued here.  if we already saw 'async' once, then certainly the second should not be treated as a modifier right?

Reply:
According to initial logic, the last `async` was considered an identifier, while all `async`s before as modifiers. There were tests like `async async async` and the expected behaviour was _modifier modifier identifier_. In case of single `async` it was treated as _identifier_. My initial intention was to change only this case.